### DISCUSSION TITLE
Deleting Received Message for Us Only

### DISCUSSION THREAD

**Discussion by ghafarizar** [tulir/whatsmeow#397] [WhatsApp protocol Q&A]
How to delete chats from client?
The case is when someone using my Bot to process something secret (documents, etc.)
As far as i know, revoke only for our own chat.
Im migrating from Nodejs to Golang.
and nodejs library "Whatsapp-Web.js" for example, can delete our chat and client chat too
I think thats achievable right?
or is now achieveable ? if so, please tell me the example...
thankyou!

**Comment by mamur-rezeki** [comment]
```go

// Needed fields
var isme = client.Store.ID.ToNonAD().String() == *ctx.Participant
var id_message = ctx.StanzaId
var participant_sender = ctx.Participant


var delete_protocol =  &waProto.Message{
	ProtocolMessage: &waProto.ProtocolMessage{
		Key: &waProto.MessageKey{
			FromMe:      &isme,
			RemoteJid:   proto.String(event.Info.Chat.ToNonAD().String()),
			Id:          id_message,
			Participant: participant_sender,
		},
		Type: waProto.ProtocolMessage_REVOKE.Enum(),
	},
}
```

The `ctx` context  is obtained from the quoted message that needs to be deleted. Therefore, it is important here to obtain the 3 fields mentioned above.

`isme` Result of perform a check to determine if the ID (JID) present in the message context to be deleted is the same as our Bot's ID

**Reply by ghafarizar** [reply]
i still cannot find the participant object
My process will stuck if its trying to send this ProtocolMessage

``` 
	client.SendMessage(context.Background(), types.JID{
		User:   v.Info.Sender.User,
		Server: types.DefaultUserServer,
	}, &waProto.Message{
		ProtocolMessage: &waProto.ProtocolMessage{
			Key: &waProto.MessageKey{
				FromMe:      &v.Info.IsFromMe,
				RemoteJid:   proto.String(v.Info.Chat.ToNonAD().String()),
				Id:          v.Message.Chat.Id,
			},
			Type: waProto.ProtocolMessage_REVOKE.Enum(),
		},
	})

**Reply by mamur-rezeki** [reply]
`Participant` field should be filled. It filled with jid which user that has send the message or who owner of the message that we want to be deleted.

So if we want to delete incoming message, we can put `event.Info.Sender.ToNonAD().String()`
